src/llama.cpp

```diff
@@ -6942,7 +6951,14 @@ static bool llm_load_tensors(
 
     // assign cpu layers
     for (int i = 0; i < i_gpu_start; ++i) {
+#ifdef GGML_USE_AMX
+        model.buft_layer[i] = {
+            ggml_backend_amx_buffer_type(),
+            llama_default_buffer_type_cpu(model, true)
+        };
+#else
         model.buft_layer[i] = llama_default_buffer_type_cpu(model, true);
+#endif
```

Comment:
This doesn't need to be changed now, but just as a heads ups, I will change the way this is done completely in a future refactor. I do not have a machine with AMX to test this backend, so it will be important to have CI tests for AMX.

Reply:
covered in https://github.com/ggerganov/llama.cpp/pull/7707#issuecomment-2275136164